src/shared/UniversalRender.jsx

```diff
@@ -45,7 +45,12 @@ const calcOffsetRoot = (startEl) => {
 const SCROLL_TOP_TRIES = 50;
 const SCROLL_TOP_DELAY_MS = 100;
 const SCROLL_TOP_EXTRA_PIXEL_OFFSET = 3;
-const SCROLL_UP_FUDGE_PIXELS = 10;
+const SCROLL_FUDGE_PIXELS = 10;
```

Comment:
this logic is highly complex. can you please add some comments to describe the control flow in plain English?

Reply:
👍